### ISSUE TITLE
otomi validate-templates stuck at drone/templates/secrets.yaml

### ISSUE THREAD

**Issue by svatwork** [redkubes/otomi-core#291]
**Describe the bug:** (a clear and concise description of what the bug is)

When executing `otomi validate-templates` with the latest pull from master, the following error is thrown:

```bash
helmfile-init.yaml: basePath=.
helmfile-03.init.yaml: basePath=.
in helmfile.d/helmfile-30.admin.yaml: command "/home/app/tools/helm" exited with non-zero status:

PATH:
  /home/app/tools/helm

ARGS:
  0: helm (4 bytes)
  1: template (8 bytes)
  2: drone (5 bytes)
  3: ../charts/drone (15 bytes)
  4: --namespace (11 bytes)
  5: team-admin (10 bytes)
  6: --values (8 bytes)
  7: /tmp/values670743084 (20 bytes)
  8: --output-dir (12 bytes)
  9: /tmp/otomi/kubeval-fixtures/helmfile-30.admin-d5e80de1-drone (60 bytes)

ERROR:
  exit status 1

EXIT STATUS
  1

STDERR:
  Error: template: drone/templates/secrets.yaml:34:106: executing "drone/templates/secrets.yaml" at <b64enc>: invalid value; expected string
  Use --debug flag to render out invalid YAML

COMBINED OUTPUT:
  Error: template: drone/templates/secrets.yaml:34:106: executing "drone/templates/secrets.yaml" at <b64enc>: invalid value; expected string
  Use --debug flag to render out invalid YAML
Processing CRD files.
Validating resources against Kubernetes version: v1.18
WARN - knative-init/templates/all.yaml containing a CustomResourceDefinition (knativeeventings.operator.knative.dev) was not validated against a schema
WARN - knative-init/templates/all.yaml containing a CustomResourceDefinition (knativeservings.operator.knative.dev) was not validated against a schema
WARN - raw/templates/resources.yaml contains an invalid ClusterIssuer (letsencrypt-staging) - spec.acme.solvers.0.dns01: Invalid type. Expected: object, given: null
WARN - raw/templates/resources.yaml contains an invalid ClusterIssuer (letsencrypt-production) - spec.acme.solvers.0.dns01: Invalid type. Expected: object, given: null
Validation Success
```

**To Reproduce**
1. Pull master
2. Run `otomi validate-templates`
3. See error

**Expected behavior:** (a clear and concise description of what you expected to happen)

```
Validation Success
```

**Comment by svatwork** [comment]
I would like to investigate further, but before I do that, maybe someone knows what's behind this?

**Comment by rawc0der** [comment]
Hmm, I saw these kinds of errors before. It happens only when helmfile does secret decryption and something did not run correctly (timeouts, unreachable urls, etc), making the templates fail. But I am not sure exactly if that's the case here.

Also the issue for exiting the scripts correctly on errors is resolved in #177.

**Comment by Morriz** [comment]
Strange, as I don't get that error. Can you reproduce @svatwork with these commands?
```
export export CLOUD=azure CLUSTER=dev # will set the bootstrap to target master
otomi bootstrap # was working before without CLOUD & CLUSTER, will create bug report
otomi validate-values
```

You should see this output:
```
Generating Kubernetes v1.18 Manifests for aws-demo.
helmfile-init.yaml: basePath=.
helmfile-03.init.yaml: basePath=.
Processing CRD files.
Validating resources against Kubernetes version: v1.18
WARN - knative-init/templates/all.yaml containing a CustomResourceDefinition (knativeeventings.operator.knative.dev) was not validated against a schema
WARN - knative-init/templates/all.yaml containing a CustomResourceDefinition (knativeservings.operator.knative.dev) was not validated against a schema
Generating Kubernetes v1.18 Manifests for aws-dev.
helmfile-init.yaml: basePath=.
helmfile-03.init.yaml: basePath=.
Processing CRD files.
Validating resources against Kubernetes version: v1.18
WARN - knative-init/templates/all.yaml containing a CustomResourceDefinition (knativeeventings.operator.knative.dev) was not validated against a schema
WARN - knative-init/templates/all.yaml containing a CustomResourceDefinition (knativeservings.operator.knative.dev) was not validated against a schema
Generating Kubernetes v1.19 Manifests for azure-demo.
helmfile-init.yaml: basePath=.
helmfile-03.init.yaml: basePath=.
Processing CRD files.
Validating resources against Kubernetes version: v1.19
WARN - knative-init/templates/all.yaml containing a CustomResourceDefinition (knativeeventings.operator.knative.dev) was not validated against a schema
WARN - knative-init/templates/all.yaml containing a CustomResourceDefinition (knativeservings.operator.knative.dev) was not validated against a schema
Generating Kubernetes v1.18 Manifests for azure-dev.
helmfile-init.yaml: basePath=.
helmfile-03.init.yaml: basePath=.
Processing CRD files.
Validating resources against Kubernetes version: v1.18
WARN - knative-init/templates/all.yaml containing a CustomResourceDefinition (knativeeventings.operator.knative.dev) was not validated against a schema
WARN - knative-init/templates/all.yaml containing a CustomResourceDefinition (knativeservings.operator.knative.dev) was not validated against a schema
Generating Kubernetes v1.18 Manifests for google-dev.
helmfile-init.yaml: basePath=.
helmfile-03.init.yaml: basePath=.
Processing CRD files.
Validating resources against Kubernetes version: v1.18
WARN - knative-init/templates/all.yaml containing a CustomResourceDefinition (knativeeventings.operator.knative.dev) was not validated against a schema
WARN - knative-init/templates/all.yaml containing a CustomResourceDefinition (knativeservings.operator.knative.dev) was not validated against a schema
Generating Kubernetes v1.18 Manifests for onprem-dev.
helmfile-init.yaml: basePath=.
helmfile-03.init.yaml: basePath=.
Processing CRD files.
Validating resources against Kubernetes version: v1.18
WARN - knative-init/templates/all.yaml containing a CustomResourceDefinition (knativeeventings.operator.knative.dev) was not validated against a schema
WARN - knative-init/templates/all.yaml containing a CustomResourceDefinition (knativeservings.operator.knative.dev) was not validated against a schema
Validation Success
```

I am closing this as a nopro.
